crates/superposition_types/src/config.rs

```diff
@@ -0,0 +1,273 @@
+use std::collections::{HashMap, HashSet};
+
+use derive_more::{AsRef, Deref, DerefMut, Into};
+#[cfg(feature = "diesel_derives")]
+use diesel::{deserialize::FromSqlRow, expression::AsExpression, sql_types::Json};
+use serde::{Deserialize, Deserializer, Serialize};
+use serde_json::{json, Map, Value};
+#[cfg(feature = "diesel_derives")]
+use superposition_derives::{JsonFromSql, JsonToSql};
+
+use crate::{overridden::filter_keys_by_prefix, Cac, Contextual, Exp};
+
+macro_rules! impl_try_from_map {
+    ($wrapper:ident, $type:ident, $validate:expr) => {
+        impl TryFrom<Map<String, Value>> for $wrapper<$type> {
+            type Error = String;
+
+            fn try_from(map: Map<String, Value>) -> Result<Self, Self::Error> {
+                Ok(Self($validate(map)?))
+            }
+        }
+
+        impl<'de> Deserialize<'de> for $wrapper<$type> {
+            fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+            where
+                D: Deserializer<'de>,
+            {
+                let map = Map::<String, Value>::deserialize(deserializer)?;
+                Self::try_from(map).map_err(serde::de::Error::custom)
+            }
+        }
+
+        impl $wrapper<$type> {
+            pub fn try_from_db(map: Map<String, Value>) -> Result<Self, String> {
+                #[cfg(feature = "disable_db_data_validation")]
+                return Ok(Self($type(map)));
+                #[cfg(not(feature = "disable_db_data_validation"))]
+                return Self::try_from(map);
+            }
+        }
+    };
+}
+
+#[cfg(feature = "diesel_derives")]
+#[derive(
+    Deserialize,
+    Serialize,
+    Clone,
+    AsRef,
+    Deref,
+    DerefMut,
+    Debug,
+    Eq,
+    PartialEq,
+    Into,
+    AsExpression,
+    FromSqlRow,
+    JsonFromSql,
+    JsonToSql,
+)]
+#[diesel(sql_type = Json)]
+pub struct Overrides(Map<String, Value>);
+
+#[cfg(not(feature = "diesel_derives"))]
+#[derive(
+    Deserialize, Serialize, Clone, AsRef, Deref, DerefMut, Debug, Eq, PartialEq, Into,
+)]
+pub struct Overrides(Map<String, Value>);
+
+impl Overrides {
+    fn validate_data(override_map: Map<String, Value>) -> Result<Self, String> {
+        if override_map.is_empty() {
+            log::error!("Override validation error: Override is empty");
+            return Err("Override should not be empty".to_owned());
+        }
+        Ok(Self(override_map))
+    }
+}
+
+impl IntoIterator for Overrides {
+    type Item = (String, Value);
+    type IntoIter = <Map<String, Value> as IntoIterator>::IntoIter;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.0.into_iter()
+    }
+}
+
+impl_try_from_map!(Cac, Overrides, Overrides::validate_data);
+impl_try_from_map!(Exp, Overrides, Overrides::validate_data);
+
+#[cfg(feature = "diesel_derives")]
```

Comment:
`diesel_derives` is added to derive `JsonFromSql` and `JsonToSql` from `superposition_derives` on `Overrides` and `Condition` so that these types can directly be used in `Context` db type in models (added this behind a feature to avoid mandatory inclusion of `diesel` when importing `superposition_types`